src/components/controller.cpp

```diff
@@ -530,6 +530,25 @@ bool controller::forward_action(const actions_util::action& action_triple) {
   return true;
 }
 
+void controller::switch_module_visibility(string module_name_raw, int visible) {
+  for (auto&& mod : m_modules) {
+    if (mod->name_raw() != module_name_raw)
+      continue;
+
+    if (visible == 0) {
+      mod->set_visible(false);
+    } else if (visible == 1) {
+      mod->set_visible(true);
+    } else if (visible == 2) {
+      mod->set_visible(!mod->visible());
+    }
+
```

Comment:
I'm gonna elect to use `broadcast` because that minimizes the calls to the signal emitter, which would be (slightly) helpful if removing the signal system would be an eventuality.